### ISSUE TITLE
Sp2 Immutable Fields Discussion

### ISSUE THREAD

**Issue by tkiethanom** [oyiptong/splice#185]

**Comment by ncloudioj** [comment]
Following are all the **mutable** fields we have, note that it's not yet the finally edition. @tspurway @rlr can you guys review it please? 

For the `adgroups` and `tiles`, we have a less strict criterion than the one in the current Splice, which treats every single change in the adgroup and tile as a new entry in the database. For entries with a trailing question mark, particularly in the `adgroups`, those are not 100% sure fields for me. 

# Accounts
* name
* contact_name
* contact_email
* contact_phone

# Campaigns:
* name ?
* countries
* channel_id
* paused

# Adgroups
* name?
* type, [directory | suggested] ?
* paused
* frequency_cap_daily ?
* frequency_cap_total ?
* categories ?

# Tiles
* status, [approved | disapproved | unapproved]
* paused
* bg_color
* title_bg_color

**Comment by tkiethanom** [comment]
Don't we need to be able to change the end_date of a campaign in case you want to stop it?

**Comment by ncloudioj** [comment]
Yes, that's good point. Most likely, we want them to be mutable.

They're also pretty tricky though, if we allow user to change them, the index generating API may see strange things. For example, change the start_date from two month ago to today.

**Comment by rlr** [comment]
Maybe you can only change start or end if it is in the future. And you can't change them to the past. ew complexity

**Comment by tkiethanom** [comment]
@ncloudioj is Ad Group explanation supposed to be immutable?

**Comment by ncloudioj** [comment]
@tkiethanom yes, the explanation is immutable.
